src/main/java/com/iota/iri/model/HashFactory.java

```diff
@@ -0,0 +1,153 @@
+        
+        try {
+            return constructor.newInstance(source, sourceOffset, sourceSize);
+        } catch(RuntimeException | InstantiationException | IllegalAccessException | InvocationTargetException e) {
+            return Hash.NULL_HASH;
+        }
+    }
+    
+    /**
+     * Creates a Hash using the provided trits
+     * @param trits the source data
+     * @param sourceOffset the offset we start reading from
+     * @return the hash
+     */
+    public Hash create(String trits) {
+        if (stringConstructor == null) return Hash.NULL_HASH;
+      
+        try {
+            return stringConstructor.newInstance(trits);
+        } catch(RuntimeException | InstantiationException | IllegalAccessException | InvocationTargetException e) {
+            return Hash.NULL_HASH;
+        }
+    }
+
+    /**
+     * Creates a Hash using the provided trits
+     * @param trits the source data
+     * @param sourceOffset the offset we start reading from
+     * @param sourceSize the size this hash is in bytes, starting from offset
+     * @return the hash
+     */
+    public Hash create(byte[] source, int sourceOffset, int sourceSize) {
+        return createFromReflection(source, sourceOffset, sourceSize);
+    }
+    
+    /**
+     * Creates a Hash using the provided trits
+     * @param trits the source data
+     * @param sourceOffset the offset we start reading from
+     * @return the hash
+     */
+    public Hash create(byte[] trits, int sourceOffset) {
+        return createFromReflection(trits, sourceOffset, Hash.SIZE_IN_TRITS);
+    }
+    
+    /**
+     * Creates a Hash using the provided source.
+     * Starts from the beginning, source size is based on source length
+     * @param source the source data
+     * @return the hash
+     */
+  	public Hash create(byte[] source) {
+  		return createFromReflection(source, 0, source.length == Hash.SIZE_IN_TRITS ? Hash.SIZE_IN_TRITS : Hash.SIZE_IN_BYTES);
+  	}
+  	
+  	//
+  	// These are used by RocksDB
+  	//
+
+    /**
+     * 
+     * @param modelClass The model this Hash represents
+     * @param source the source data, bytes or trits. Based on the length of source data
+     * @return the hash of the correct type
+     */
+    public Hash create(Class<?> modelClass, byte[] source) {
+        return create(modelClass, source, 0, source.length == AbstractHash.SIZE_IN_TRITS ? AbstractHash.SIZE_IN_TRITS : AbstractHash.SIZE_IN_BYTES);
+    }
+
+    /**
+     * 
+     * @param modelClass The model this Hash represents
+     * @param source the source data, bytes or trits
+     * @param sourceOffset the offset in the source
+     * @param sourceSize the size this hash is in bytes, starting from offset
+     * @return the hash of the correct type
+     */
+    public Hash create(Class<?> modelClass, byte[] source, int sourceOffset, int sourceSize) {
+        //Transaction is first since its the most used
+        if (modelClass.equals(Transaction.class)) {
```

Comment:
As a general rule of thumb It is better to avoid null pointer exceptions by doing either:
`Transaction.class.equals(modelClass)` or `Objects.equals(modelClass, Transaction.class)`.
The latter is better when you have 2 variables that each can be null.

The exception to the rule is when you want to throw null pointer exceptions, but in most cases you don't.
I always do equality this way even when the variable can't be null just to avoid the risk and as a force of habit.